src/registry/map/preview.tsx

```diff
@@ -94,6 +99,14 @@ const Preview: React.FC<ComponentPreviewProps<MapComponentSchema>> = ({component
     return <Loader />;
   }
 
+  try {
+    zodSchema({intl, builderContext}).parse(component);
+  } catch (error) {
+    if (error instanceof z.ZodError) {
+      throw error;
+    }
+  }
```

Comment:
```suggestion
  const {success: isValidConfiguration} = zodSchema({intl, builderContext}).safeParse(component)
  
  if (!isValidConfiguration) {
    return <FormattedMessage description="Map preview: invalid config message" defaultMessage="The map configuration is not valid, so we can't show a preview. Fix the validation errors in the component configuration." />;
  };
```
Something like that :)